components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java

```diff
@@ -90,7 +95,325 @@ public class FIDOAuthenticator extends AbstractApplicationAuthenticator
     public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletResponse response,
                                            AuthenticationContext context)
             throws AuthenticationFailedException, LogoutFailedException {
-        return super.process(request, response, context);
+
+        // Extract the configurations
+        boolean enablePasskeyProgressiveEnrollment = isPasskeyProgressiveEnrollmentEnabled(context.getTenantDomain());
+        boolean enableUsernamelessAuthentication = isUsernamelessAuthenticationEnabled(context.getTenantDomain());
+        addPasskeyEnrollmentConfigToEndpointParams(context, enablePasskeyProgressiveEnrollment);
+
+        // If the logout request comes, then no need to go through and complete the flow.
+        if (context.isLogoutRequest()) {
+            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+        }
+
+        // If an authentication complete request comes go through this flow.
+        if (StringUtils.isNotEmpty(request.getParameter(TOKEN_RESPONSE)) &&
+                !(!StringUtils.isEmpty(request.getParameter(SCENARIO)) &&
+                        request.getParameter(SCENARIO).equals(ScenarioTypes.INIT_FIDO_ENROL))) {
+            processAuthenticationResponse(request, response, context);
+            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+        }
+
+        // If a passkey enrollment request comes set a property to the context mentioning the user consent is received.
+        if (enablePasskeyProgressiveEnrollment && !StringUtils.isEmpty(request.getParameter(SCENARIO)) &&
+                request.getParameter(SCENARIO).equals(ScenarioTypes.INIT_FIDO_ENROL)) {
```

Comment:
Let's switch the constant to avoid any possible NPEs.

Reply:
Resolved via [59395d7](https://github.com/wso2-extensions/identity-local-auth-fido/pull/110/commits/59395d7a0c1dad27440c38d7ed768bb016868b68)